### ISSUE TITLE
[s2n] build failure

### ISSUE THREAD

**Issue by csw** [microsoft/vcpkg#21882]
**Host Environment**
- OS: Arch Linux
- Compiler: gcc 11.1.0

**To Reproduce**
`./vcpkg install s2n`

**Failure logs** 

```
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /home/csw/src/vcpkg/buildtrees/s2n/x64-linux-dbg
    See logs for more information:
      /home/csw/src/vcpkg/buildtrees/s2n/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:102 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/s2n/portfile.cmake:14 (vcpkg_cmake_install)
  scripts/ports.cmake:142 (include)


Error: Building package s2n:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+s2n
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[s2n]+Build+error
including:
  package: s2n[core]:x64-linux -> 1.1.1
    vcpkg-tool version: 2021-11-24-48b94a6946b8a70abd21529218927fd478d02b6c
    vcpkg-scripts version: 63e935d96 2021-12-02 (4 days ago)
```

Full log is in [install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/7662673/install-x64-linux-dbg-out.log), but because s2n is built with -Werror by default, it fails with
```
/home/csw/src/vcpkg/buildtrees/s2n/src/e46cc6454b-8c9ecf3bde.clean/pq-crypto/kyber_90s_r2/ntt.c:92:44: error: argument 1 of type ‘int16_t[256]’ {aka ‘short int[256]’} with mismatched bound [-Werror=array-parameter=]
   92 | void PQCLEAN_KYBER51290S_CLEAN_ntt(int16_t poly[256]) {
      |                                    ~~~~~~~~^~~~~~~~~
```

**Additional context**

I resolved this in https://github.com/Genomenon/vcpkg/commit/8869cd0d2b0dae4267358867d84d978f0de28c67 by updating s2n to version 1.3.1 and building with `-DUNSAFE_TREAT_WARNINGS_AS_ERRORS=OFF`, since compiler warnings like this are not actionable for vcpkg users. I can make a corresponding PR if this seems like the right approach.

**Comment by dg0yt** [comment]
A more elaboraed s2n update is waiting in #21484. It is only version 1.3.0. Stuck in review due to offering self-tests as feature.

**Comment by JackBoosY** [comment]
Will fix this issue after #21484 merge.

**Comment by csw** [comment]
Just verified that this is still a problem with the 1.3.0 update. The upstream issue is https://github.com/aws/s2n-tls/issues/2989; pending a fix for that, I think `-DUNSAFE_TREAT_WARNINGS_AS_ERRORS=OFF` will be required to make it build with gcc 11.
